src/common/buffer.cc

```diff
@@ -1808,6 +1810,27 @@ void buffer::list::write_stream(std::ostream &out) const
   }
 }
 
+void buffer::list::compress(list& dest)
+{
+  const char* pch_src = c_str();
+  uint32_t input_size = length();
+  uint32_t max_compressed_size = LZ4_compressBound(input_size);
+  char *pch_dest = new char[max_compressed_size];
+  uint32_t bytes_passed = LZ4_compress(pch_src, pch_dest, input_size);
+  dest.append(pch_dest, bytes_passed);
```

Comment:
this allocates, decompresses, copies, then deallocates.  Instead, you can do

 bufferptr bp = buffer::create_page_aligned(max_compressed_size);
 uint32_t actual = LZ4_compress(pch_src, bp.c_str(), input_size);
 bp.set_len(actual);
 dest.append(bp);
to avoid the copy.  we will allocate more memory than we need, which isn't ideal, but probably good enough for now.